src/effects/AutoDuck.cpp

```diff
@@ -195,12 +195,14 @@ bool EffectAutoDuck::Process(EffectInstance &, EffectSettings &)
       const auto t0 = pControlTrack->LongSamplesToTime(start);
       const auto t1 = pControlTrack->LongSamplesToTime(end);
       if (WaveTrackUtilities::HasStretch(*pControlTrack, t0, t1)) {
-         // TODO a progress indicator?
-         tempTracks = pControlTrack->Duplicate({ { t0, t1 } });
+         tempTracks = pControlTrack->Duplicate();
          if (tempTracks) {
             const auto pFirstTrack = *tempTracks->Any<WaveTrack>().begin();
-            if (pFirstTrack)
+            if (pFirstTrack) {
+               // TODO a progress indicator?
```

Comment:
Maybe the progress indicator for stretch can be built by a reusable utility function in WaveTrackUtils.h.  It would take another std::function just to report the varying `count` and give back another std::function.

Reply:
I'm doing something like that in 4-of-6, where rendering happens also e.g. when joining.

Reply:
On further thought, the new utility function could just take a reference to a counter variable instead of a function.  It would return a function.

Of course you would need to use C++14 "extended capture" [] syntax to allocate the progress indicator with `make_unique` so it would be destroyed along with the `std::function` when no longer needed.